artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContext.java

```diff
@@ -128,10 +147,11 @@ public void run() {
          }
 
          connection.requireInHandler();
-         if (receiver.getCredit() <= threshold) {
-            int topUp = refill - receiver.getCredit();
+         int pending = context != null ? context.pendingSettles : 0;
```

Comment:
@tabish121 also, in case you mix durable and non durable and use the executors, this will adjust pending settles on the credits.